libsql-server/src/auth/user_auth_strategies/jwt.rs

```diff
@@ -7,8 +7,13 @@ use crate::{
 
 use super::{UserAuthContext, UserAuthStrategy};
 
+pub enum DecodingKeyContainer {
```

Comment:
It won't be long until we forget why we implemented this single/multiple distinction
it is a seed of complexity -> it requires forked logic to handle
any chance we could have one more round of conversations about how much real overhead a vec! introduces?

Reply:
I'm okay removing the complexity if everyone prefers that after looking at the enum version. Could be a classic case of premature optimization. Fwiw, I do think it should remaining somewhat intuitive that the vast majority of the time there should only be a single key and hence a motivation for this bit of weirdness.

Reply:
let's let the others chime in then

Reply:
I second @shopifyski here, I'd rather have a single code path, the overhead of allocating a vector once in a while is negligible

Reply:
As an extra data point, I was exploring today what it would look like with just `Vec<jsonwebtoken::DecodingKey>` everywhere except for the `DecodingKeyContainer` enum. The immediate instinct was to rename all `jwt_key` to `jwt_keys` since it felt weird to do things like `pub fn parse_jwt_key(data: &str) -> Result<Vec<DecodingKey>>` or `let key: Vec<DecodingKey> = ...`. However, as the thread kept on pulling I started to run into some backwards-compatibility issues, specifically with the `jwt_key` (singular) request variable in libsql-server/src/http/admin/mod.rs. We can certainly halt the cascade of singular-to-plural changes anywhere we like, but at some point we may have something gross like `config.jwt_keys = req.jwt_key`.

Reply:
We could extend the config with `jwt_keys`, and merge `jwt_key` and `jwt_keys` as late as possible when building the decoding object. We would mark the `jwt_key` field as deprecated, and cleanup sometime in the future, would that contain the changes?

Reply:
I opted to keep the stuff outside of libsql-server as is. It didn't feel as bizarre as I had feared to draw the line here.